src/main/java/org/example/bookstore/controller/BookControllerImpl.java

```diff
@@ -17,36 +19,44 @@
 import org.springframework.web.bind.annotation.ResponseStatus;
 import org.springframework.web.bind.annotation.RestController;
 
-@Component
 @RestController
-@AllArgsConstructor
+@RequiredArgsConstructor
 @RequestMapping(value = "/api/books")
 public class BookControllerImpl {
 
     private final BookService bookService;
 
     @GetMapping
-    public List<BookDto> getAll() {
-        return bookService.findAll();
+    @Operation(summary = "Get all books", description = "Get all existing books from DB")
+    public List<BookDto> getAllBooks(Pageable pageable) {
+        return bookService.findAll(pageable);
     }
 
     @GetMapping("/{id}")
+    @Operation(summary = "Get book by id", description = "Get existing book by id")
     public BookDto getBookById(@PathVariable Long id) {
         return bookService.getBookById(id);
     }
 
     @PostMapping
-    public BookDto createBook(@RequestBody BookRequestDto requestDto) {
+    @ResponseStatus(HttpStatus.CREATED)
+    @Operation(summary = "Create book", description = "Create book with parameters in DB")
+    public BookDto createBook(@RequestBody @Valid BookRequestDto requestDto) {
```

Comment:
we can also describe the request DTO itself with it's fields, for example note a format of ISBN that we accept, which is usually the most useful kind of information when it comes to working in the team